demisto_sdk/commands/lint/linter.py

```diff
@@ -1306,3 +1337,280 @@ def _get_commands_list(self, script_obj: dict):
+        """
+        Gets a specific native image name, and finds it's reference (tag) in the docker_native_image_config.json
+        file. If the native image name doesn't exist in the file or the integration/script that lint runs on doesn't
+        support this native image - write to the logs that the checks on docker will be skipped.
+        Args:
+            native_image (str): Name of the Native image to run on.
+            native_image_config (NativeImageConfig): NativeImageConfig obj based on the docker_native_image_config.json.
+            supported_native_images (set): A set including the names of the native images supported for the
+                                           script/integration that lint runs on.
+            script_id (str): The ID of the integration/script that lint runs on.
+        Returns (str): The native image reference (tag).
+        """
+        log_prompt = f"{self._pack_name} - Get Specific Native Image"
+        logger.info(f"{log_prompt} - {native_image} - Started")
+
+        if native_image not in native_image_config.native_images:
+            # Server version is invalid or not exist in the docker_native_image_config.json
+            logger.error(
+                f"{log_prompt} - Skipping checks on docker for {native_image}. The requested native image: "
+                f"{native_image} is not supported. For supported native image versions please see: "
+                f"'Tests/docker_native_image_config.json'"
+            )
+            # TODO: Maybe should throw an exception instead
+            return None
+
+        elif native_image not in supported_native_images:
+            # Integration/Script doesn't support the requested native image
+            logger.info(
+                f"{log_prompt} - Skipping checks on docker for {native_image} - {script_id} does not support the "
+                f"requested native image: {native_image}."
+            )
+            return None
+
+        else:
+            # Integration/Script supports the requested native image - find the relevant tag to run on
+            native_image_reference = native_image_config.get_native_image_reference(
+                native_image
+            )
+
+            return native_image_reference
+
+    def _get_docker_images_from_yml(self, script_obj: Dict) -> List[str]:
+        """
+        Gets a yml as dict of the script/integration that lint runs on, and returns a list of all 'dockerimage' values
+        in the yml (including 'alt_dockerimages' if the key exist).
+        Args:
+            script_obj (dict): A yml as dict of the integration/script that lint runs on.
+        Returns (List): A list including all the docker images of the integration/script.
+        """
+        log_prompt = f"{self._pack_name} - Get Docker Images From YML"
+        logger.info(f"{log_prompt} - Started")
+
+        imgs = get_all_docker_images(script_obj)
+
+        return imgs
+
+    def _get_all_docker_images(
```

Comment:
looks like this function is doing a lot of duplicate work. We should add the `all` case as part of the if then flow in ```
_get_docker_images_for_lint instead. eg
```
images = []
if flag==all or flag==latest 
     images += latest
if flag == all or flag == from-yml
    images += fromyml
return images    

```

Reply:
@JudahSchwartz - I've tried it, but I think it will end up being more cumbersome, making it harder to write unit tests and read the code.